sdks/python/apache_beam/runners/dataflow/internal/apiclient.py

```diff
@@ -506,7 +506,13 @@ def __init__(self, options, proto_pipeline):
     # Labels.
     if self.google_cloud_options.labels:
       self.proto.labels = dataflow.Job.LabelsValue()
-      for label in self.google_cloud_options.labels:
+      labels = self.google_cloud_options.labels
+      if len(labels) == 1:
```

Comment:
Could you please add a test for this here - https://github.com/apache/beam/blob/380359ce9ebd284ecd9bc4c91efdd5f30da21425/sdks/python/apache_beam/runners/dataflow/internal/apiclient_test.py#L786?